rllib/utils/schedules/polynomial_schedule.py

```diff
@@ -0,0 +1,46 @@
+from ray.rllib.utils.schedules.schedule import Schedule
+from ray.rllib.utils.framework import try_import_tf
+
+tf = try_import_tf()
+
+
+class PolynomialSchedule(Schedule):
+    def __init__(self,
+                 schedule_timesteps,
+                 final_p,
+                 initial_p=1.0,
+                 power=2.0,
+                 framework=None):
+        """
+        Polynomial interpolation between initial_p and final_p over
+        schedule_timesteps. After this many time steps always `final_p` is
+        returned.
+
+        Agrs:
+            schedule_timesteps (int): Number of time steps for which to
+                linearly anneal initial_p to final_p
+            final_p (float): Final output value.
+            initial_p (float): Initial output value.
+            framework (Optional[str]): One of "tf", "torch", or None.
+        """
+        super().__init__(framework=framework)
+        assert schedule_timesteps > 0
+        self.schedule_timesteps = schedule_timesteps
+        self.final_p = final_p
+        self.initial_p = initial_p
+        self.power = power
+
+    def value(self, t):
+        """
+        Returns the result of:
+        final_p + (initial_p - final_p) * (1 - `t`/t_max) ** power
+        """
+        if self.framework == "tf" and tf.executing_eagerly() is False:
+            return tf.train.polynomial_decay(
```

Comment:
I'm wondering if schedules should support frameworks. We could always call them in Python land and feed their values in, which would make them framework independent automatically (as they are currently I believe).

Reply:
Basically, instead of feeding in a `t` tensor and letting the schedule be computed in-graph we feed in the `value` tensor (after computing the schedule out of graph).

Reply:
The problem is when stuff happens deeper inside some components. How is Policy supposed to know that its Exploration object (which could be of any type) needs some Schedule's value (or even know that the Exploration object has a Schedule (not all Explorations do))?

Reply:
An example, where a Schedule could be really buried in the middle of the graph: In SAC discrete, if you were to use a decaying temperature parameter for your Gumbel Softmax distribution, the sampling forward pass through the Gumbel is right in the middle between two NN forward passes (the policy NN to sample, then the Q-nets to get the action-values), which together as a whole have to be differentiable. Sure, you could still get the current decayed weight value and then pass it in, but it would make everything much more complicated for the user.

Reply:
Ok, I think there is a way we could allow Schedules to be embedded in graphs without having to do too much work to support different frameworks.

Since we don't need to differentiate through schedules, and they are not performance critical, we could add `.tf_value(time: Tensor) -> Tensor` and `.torch_value(time: Tensor) -> Tensor` helper methods.

You can implement this straightforwardly in torch / tf eager. In TF graph mode you can use `tf.py_function` (which isn't differentiable, but can allow arbitrary python code to be run in the graph). How does this sound?

Reply:
(this would also be similar to how the Exploration interface works with diff frameworks I believe).